### ISSUE TITLE
Indexing with uint8 returns wrong value

### ISSUE THREAD

**Issue by ricardoV94** [google/jax#16836]
### Description

This bug showed up in https://github.com/pymc-devs/pytensor/issues/395

```python
import jax.numpy as jnp
import numpy as np

def subtensor(x):
    return x[np.array(128, dtype="uint8")]

subtensor(jnp.arange(200))  # Array(199, dtype=int32)
```

### What jax/jaxlib version are you using?

jax v0.4.12, jaxlib v0.4.12

### Which accelerator(s) are you using?

CPU

### Additional system info

Linux

### NVIDIA GPU info

NA

**Comment by jakevdp** [comment]
Thanks! Looks like an overflow somewhere in the index normalization logic. I'll dig in.

**Comment by jakevdp** [comment]
Actually, it's not overflow in JAX's handling, it looks like overflow in the implementation of `dynamic_slice`:
```python
import jax.numpy as jnp
from jax import lax

arr = jnp.arange(200)
i = np.uint8(128)
print(jax.lax.dynamic_slice(arr, [i], [1]))
# [199]
```

**Comment by jakevdp** [comment]
Reported upstream at https://github.com/openxla/xla/issues/4491

**Comment by jakevdp** [comment]
Fixed by https://github.com/openxla/xla/commit/4e4eff35bf9a5f8ed54fd290391bd0612f49533e. Thanks!

**Comment by ricardoV94** [comment]
Thanks for the quick fix!
